apps/web/src/middlewares/repoAuth.js

```diff
@@ -18,16 +18,19 @@ export const repoAuth = [
     }
 
     if (repository.private) {
-      const account = await repository.getAccount();
-      const currentConsumption =
-        await account.getScreenshotsCurrentConsumption();
-      const maxConsumption = await account.screenshotsMonthlyLimit();
+      const account = await Account.getAccount({
+        organizationId: repository.organizationId,
+      });
+      if (account) {
+        const screenshotsConsumptionRatio =
+          await account.getScreenshotsConsumptionRatio();
 
-      if (currentConsumption / maxConsumption >= 1.1) {
-        throw new HttpError(
-          402,
-          `Build refused for insufficient credit. Thank to upgrade Argos plan`
-        );
+        if (screenshotsConsumptionRatio && screenshotsConsumptionRatio >= 1.1) {
+          throw new HttpError(
+            402,
+            `Build refused for insufficient credit. Thank to upgrade Argos plan`
+          );
+        }
       }
```

Comment:
This code should not be part of `repoAuth`, this is not the same responsibility.

Reply:
I'll be removed from the `repoAuth` repo then. I'm sure that our users will be happy too!